src/protocol/sort/apply.rs

```diff
@@ -1,87 +1,141 @@
     use rand::seq::SliceRandom;
 
     #[test]
-    fn apply_shares() {
+    fn apply_just_one_cycle() {
         let mut values = ["A", "B", "C", "D"];
-        let indices = Permutation::oneline([2, 3, 1, 0]).inverse();
+        let permutation = [2, 3, 1, 0];
         let expected_output_apply = ["C", "D", "B", "A"];
-        apply(indices, &mut values);
+        apply(&permutation, &mut values);
         assert_eq!(values, expected_output_apply);
+    }
 
+    #[test]
+    fn apply_just_two_cycles() {
+        let mut values = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"];
+        let permutation = [3, 4, 6, 7, 0, 9, 10, 1, 2, 8, 5];
+        let expected_output_apply = ["D", "E", "G", "H", "A", "J", "K", "B", "C", "I", "F"];
+        apply(&permutation, &mut values);
+        assert_eq!(values, expected_output_apply);
+    }
+
+    #[test]
+    fn apply_complex() {
+        let mut values = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"];
+        let permutation = [1, 0, 2, 5, 6, 7, 8, 9, 10, 3, 4];
+        let expected_output_apply = ["B", "A", "C", "F", "G", "H", "I", "J", "K", "D", "E"];
+        apply(&permutation, &mut values);
+        assert_eq!(values, expected_output_apply);
+    }
+
+    #[test]
+    fn apply_inv_just_one_cycle() {
         let mut values = ["A", "B", "C", "D"];
-        let indices = Permutation::oneline([2, 3, 1, 0]).inverse();
-        let expected_output_apply_inv = ["D", "C", "A", "B"];
-        apply_inv(indices, &mut values);
-        assert_eq!(values, expected_output_apply_inv);
+        let permutation = [2, 3, 1, 0];
+        let expected_output_apply = ["D", "C", "A", "B"];
+        apply_inv(&permutation, &mut values);
+        assert_eq!(values, expected_output_apply);
     }
 
     #[test]
-    pub fn composing() {
-        let sigma = vec![4, 2, 0, 5, 1, 3];
-        let mut rho = vec![3, 4, 0, 5, 1, 2];
+    fn apply_inv_just_two_cycles() {
+        let mut values = ["A", "B", "C", "D", "E"];
+        let permutation = [3, 4, 1, 0, 2];
+        let expected_output_apply = ["D", "C", "E", "A", "B"];
+        apply_inv(&permutation, &mut values);
+        assert_eq!(values, expected_output_apply);
+    }
 
-        // Applying sigma on rho
-        apply_inv(Permutation::oneline(sigma), &mut rho);
-        assert_eq!(rho, vec![1, 0, 3, 2, 4, 5]);
+    #[test]
+    fn apply_inv_complex() {
+        let mut values = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"];
+        let permutation = [1, 0, 2, 5, 6, 7, 8, 9, 10, 3, 4];
+        let expected_output_apply = ["B", "A", "C", "J", "K", "D", "E", "F", "G", "H", "I"];
+        apply_inv(&permutation, &mut values);
+        assert_eq!(values, expected_output_apply);
     }
 
     #[test]
-    pub fn apply_apply_inv_relation() {
-        // This test shows that apply(permutation, values) is same as apply_inv(permutation.inverse(), values)
-        let batchsize: usize = 100;
+    fn permutations_a_million_long() {
```

Comment:
I don't know...do you really need to sort a billion here? this test validates for correctness, so maybe a million would do just fine (16 ms instead of 2 secs) :)